devito/__init__.py

```diff
@@ -102,7 +102,7 @@ def reinit_compiler(val):
 configuration.add('safe-math', 0, [0, 1], preprocessor=bool, callback=reinit_compiler)
 
 # Enable/disable automatic padding for allocated data
-configuration.add('autopadding', False, [False, True])
+configuration.add('autopadding', False, [False, True, np.float32, np.float64])
```

Comment:
How are you passing `np.float32` as en env var? This needs to accept a string of the type or like "float32/f32"

Reply:
fixed